store/sqlstore/channel_store.go

```diff
@@ -416,6 +422,268 @@ func (s SqlChannelStore) createIndexesIfNotExists() {
 }
 
+// MigrateSidebarCategories creates 3 initial categories for all existing user/team pairs
+// **IMPORTANT** This function should only be called from the migration task and shouldn't be used by itself
+func (s SqlChannelStore) MigrateSidebarCategories(fromTeamId, fromUserId string) (map[string]interface{}, *model.AppError) {
+	var userTeamMap []struct {
+		UserId string
+		TeamId string
+		Locale *string
+	}
+
+	transaction, err := s.GetMaster().Begin()
+	if err != nil {
+		return nil, model.NewAppError("SqlChannelStore.MigrateSidebarCategories", "store.sql_channel.MigrateSidebarCategories.open_transaction.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	defer finalizeTransaction(transaction)
+
+	if _, err := transaction.Select(&userTeamMap, "SELECT TeamId, UserId, Users.Locale FROM TeamMembers LEFT JOIN Users ON Users.Id=UserId WHERE (TeamId, UserId) > (:FromTeamId, :FromUserId) ORDER BY TeamId, UserId LIMIT 100", map[string]interface{}{"FromTeamId": fromTeamId, "FromUserId": fromUserId}); err != nil {
+		return nil, model.NewAppError("SqlChannelStore.MigrateSidebarCategories", "store.sql_channel.migrate_categories.select.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	if len(userTeamMap) == 0 {
+		// No more team members in query result means that the migration has finished.
+		return nil, nil
+	}
+
+	for _, u := range userTeamMap {
+		locale := "en"
+		if u.Locale != nil {
+			locale = *u.Locale
+		}
+		T := utils.GetUserTranslations(locale)
+		if err := transaction.Insert(&model.SidebarCategory{
+			DisplayName: T("sidebar.category.favorites"),
+			Id:          model.NewId(),
+			UserId:      u.UserId,
+			TeamId:      u.TeamId,
+			Sorting:     model.SidebarCategorySortManual,
+			SortOrder:   model.DefaultSidebarSortOrderFavorites,
+			Type:        model.SidebarCategoryFavorites,
+		}); err != nil && !IsUniqueConstraintError(err, []string{"UserId"}) {
+			return nil, model.NewAppError("SqlChannelStore.MigrateSidebarCategories", "store.sql_channel.MigrateSidebarCategories.app_error", nil, err.Error(), http.StatusInternalServerError)
+		}
+
+		if err := transaction.Insert(&model.SidebarCategory{
+			DisplayName: T("sidebar.category.channels"),
+			Id:          model.NewId(),
+			UserId:      u.UserId,
+			TeamId:      u.TeamId,
+			Sorting:     model.SidebarCategorySortManual,
+			SortOrder:   model.DefaultSidebarSortOrderChannels,
+			Type:        model.SidebarCategoryChannels,
+		}); err != nil && !IsUniqueConstraintError(err, []string{"UserId"}) {
+			return nil, model.NewAppError("SqlChannelStore.MigrateSidebarCategories", "store.sql_channel.MigrateSidebarCategories.app_error", nil, err.Error(), http.StatusInternalServerError)
+		}
+
+		if err := transaction.Insert(&model.SidebarCategory{
+			DisplayName: T("sidebar.category.dm"),
+			Id:          model.NewId(),
+			UserId:      u.UserId,
+			TeamId:      u.TeamId,
+			Sorting:     model.SidebarCategorySortAlphabetical,
```

Comment:
Sorry, late change since I forgot about this earlier, but can you make the DMs category default to recency sorting instead?
```suggestion
			Sorting:     model.SidebarCategorySortRecent,
```

Reply:
No problem. Will change